src/login/pam_systemd.c

```diff
@@ -1078,7 +1080,8 @@ static int register_session(
                 SessionContext *c,
                 UserRecord *ur,
                 bool debug,
-                char **ret_seat) {
+                char **ret_seat,
+                char **ret_type) {
```

Comment:
`ret_type` is only initialized when this function goes to the end. Please initialize on other successful paths.